kratos/utilities/read_materials_utility.cpp

```diff
@@ -176,80 +176,88 @@ void ReadMaterialsUtility::AssignPropertyBlock(Parameters Data)
-                temp[index] = value_variable[index];
-            CheckIfOverwritingValue(*p_prop, variable, temp);
-            p_prop->SetValue(variable, temp);
-        } else if(KratosComponents<Variable<Vector > >::Has(variable_name)) {
-            const Variable<Vector>& variable = KratosComponents<Variable<Vector>>().Get(variable_name);
-            CheckIfOverwritingValue(*p_prop, variable, value.GetVector());
-            p_prop->SetValue(variable, value.GetVector());
-        } else if(KratosComponents<Variable<Matrix> >::Has(variable_name)) {
-            const Variable<Matrix>& variable = KratosComponents<Variable<Matrix>>().Get(variable_name);
-            CheckIfOverwritingValue(*p_prop, variable, value.GetMatrix());
-            p_prop->SetValue(variable, value.GetMatrix());
-        } else if(KratosComponents<Variable<std::string> >::Has(variable_name)) {
-            const Variable<std::string>& variable = KratosComponents<Variable<std::string>>().Get(variable_name);
-            CheckIfOverwritingValue(*p_prop, variable, value.GetString());
-            p_prop->SetValue(variable, value.GetString());
-        } else {
-            KRATOS_ERROR << "Value type for \"" << variable_name << "\" not defined";
+    if (Data["Material"].Has("Variables")) {
+        Parameters variables = Data["Material"]["Variables"];
+        for (auto iter = variables.begin(); iter != variables.end(); ++iter) {
+            const Parameters value = variables.GetValue(iter.name());
+
+            std::string variable_name = iter.name();
+            TrimComponentName(variable_name);
+
+            // We don't just copy the values, we do some tyransformation depending of the destination variable
+            if (KratosComponents<Variable<double> >::Has(variable_name)) {
+                const Variable<double>& variable = KratosComponents<Variable<double>>().Get(variable_name);
+                CheckIfOverwritingValue(*p_prop, variable, value.GetDouble());
+                p_prop->SetValue(variable, value.GetDouble());
+            } else if(KratosComponents<Variable<bool> >::Has(variable_name)) {
+                const Variable<bool>& variable = KratosComponents<Variable<bool>>().Get(variable_name);
+                CheckIfOverwritingValue(*p_prop, variable, value.GetBool());
+                p_prop->SetValue(variable, value.GetBool());
+            } else if(KratosComponents<Variable<int> >::Has(variable_name)) {
+                const Variable<int>& variable = KratosComponents<Variable<int>>().Get(variable_name);
+                CheckIfOverwritingValue(*p_prop, variable, value.GetInt());
+                p_prop->SetValue(variable, value.GetInt());
+            } else if(KratosComponents<Variable<array_1d<double, 3> > >::Has(variable_name)) {
+                const Variable<array_1d<double, 3>>& variable = KratosComponents<Variable<array_1d<double, 3>>>().Get(variable_name);
+                array_1d<double, 3> temp(3, 0.0);
```

Comment:
```suggestion
                array_1d<double, 3> temp = ZeroVector(3);
```

This constructor is not available on Amatrix